### DISCUSSION TITLE
The aborted parts cannot be listed using "ListMultipartUploads"

### DISCUSSION THREAD

**Discussion by xiaonaiquan** [minio/minio#16101] [Q&A] [answered]
Hi,When I was doing a segmented upload, I used AbortMultipartUpload to abort the upload.Then I use the following code to try to get the aborted part, but I can't get the number.

ListMultipartUploadsRequest listMultipartRequest;
listMultipartRequest.WithBucket(params.bucketName.c_str());
ListMultipartUploadsOutcome listMultipartOutcome = m_pS3Client->ListMultipartUploads(listMultipartRequest);
if(listMultipartOutcome.IsSuccess())
 {
    std::cout << listMultipartOutcome.GetResult().GetUploads().size() << std::endl;
  }

**Comment by harshavardhana** [comment, marked as answer]
Only if you know the object name, we do not implement hierarchical listing on ListMultipartUploads() - they are not useful. Server automatically clears incomplete parts in 24hrs.

**Comment by xiaonaiquan** [comment]
So how do you know the abort part? MultipartId ？

**Reply by klauspost** [reply]
You abort by providing the upload id.

**Reply by xiaonaiquan** [reply]
I saved the upload id locally, but I don't know which class to use to get the unfinished part.

**Reply by harshavardhana** [reply]
You cannot download unfinished uploads, that is not part of S3 API

**Reply by xiaonaiquan** [reply]
No, I am not trying to download the unfinished file, I am trying to get the unuploaded part after aborting the segmented upload

**Reply by harshavardhana** [reply]
I don't think I follow - can you rephrase your ask?

**Reply by xiaonaiquan** [reply]
I sent an email to harsha@minio.io, you can have a look，thank you

**Reply by klauspost** [reply]
@xiaonaiquan We do not provide personal email support. Please ask your question here or refer to our [subscription service](https://min.io/pricing)

**Reply by xiaonaiquan** [reply]
@klauspost  An unexpected exit occurred during the piecewise upload of the program, which stopped the piecewise upload. 
After the program restarts, how do I get the unfinished part to continue uploading? 
Before I accidentally quit, I saved the upload id locally. I want to get the unfinished part through the upload id, but I don't know how.

**Reply by klauspost** [reply]
If you expect your program to crash you should save your state after each part.

You will however need to do part uploads pretty much manually. Some SDKs allow for that level of interactions, others don't.

If you have a crashed session, the easiest would be to delete the existing upload (AbortMultipart) and start from scratch.

**Reply by xiaonaiquan** [reply]
@klauspost  Yes, but the part that has been uploaded, the part that I uploaded manually, can it be completely merged?

**Reply by klauspost** [reply]
@xiaonaiquan So you are looking to create an object from whatever data was uploaded?

**Reply by xiaonaiquan** [reply]
@klauspost  No, I mean can the uploaded part and the continuing upload part be combined into a complete object.I'm actually using AWS-SDK.

**Reply by klauspost** [reply]
You can just continue uploading the following parts. You of course need to store your parts info returned after each part to be able to fill out the `CompleteMultipartUpload` request, which is why I suggested you store your part info each time a part has been uploaded.

**Reply by xiaonaiquan** [reply]
@klauspost  @harshavardhana Thank you, I solved the problem of program restart and resume the segmented upload.
